packages/compass-home/src/components/home.tsx

```diff
@@ -88,27 +69,7 @@ function reducer(state: State, action: Action): State {
         namespace: { ...defaultNS },
         isConnected: true,
         connectionTitle: action.connectionTitle,
-        instanceLoadingStatus: InstanceLoadedStatus.LOADING,
       };
-    case 'instance-loaded':
-      // We only want to progress to the LOADED state on the initial load
-      if (state.instanceLoadingStatus === InstanceLoadedStatus.LOADING) {
-        return {
-          ...state,
-          isDataLake: action.isDataLake,
-          instanceLoadingStatus: InstanceLoadedStatus.LOADED,
-        };
-      }
-      return state;
-    case 'instance-loaded-error':
-      if (state.instanceLoadingStatus === InstanceLoadedStatus.LOADING) {
-        return {
-          ...state,
-          errorLoadingInstanceMessage: action.errorMessage,
-          instanceLoadingStatus: InstanceLoadedStatus.ERROR,
```

Comment:
I think this error handling was a bit of a bucket for handling errors from fetching the instance information used to render the databases/collections list. It looks like we've moved this error to the instance plugin, however that plugin is only rendered on the databases/performance pages. When an error occurs when fetching collections now do we surface it?

Reply:
Yeah, so as you noticed instance plugin is now responsible for handling instance errors  and only instance errors ([this is where it happens](
https://github.com/mongodb-js/compass/pull/2640/files#diff-a251470f3d59dd2136fe6bbd4fb8d649d4b9b719857dc3c22726c0b3fe783e4aR47)). Other error handling was not lost though, it was moved to the corresponding plugins: databases workspace plugin is now handling errors when fetching databases failed ([here's the check](https://github.com/mongodb-js/compass/pull/2640/files#diff-f79a4b0b14bcc729ed2b7ddd3aa21a03dff81eea04a8923e0a72947db1806224R70-R78)) and collections workspace plugin is now handling collections fetching errors ([here](https://github.com/mongodb-js/compass/pull/2640/files#diff-7b4943b0f403559a86107ee169df3348869c5f4e3b23c79894dafb416c77dd34R50-R58)), all three are decoupled and manage their own state so can be easily moved around, reshuffled, and even be siblings (currently databases workspace plugin is a "child" of instance plugin).

It is done with a bit of code duplication right now, but I just couldn't find a better place to put this error component to and it's not that much code so for now I just copied it over. Maybe we can do something like a error boundary component in compass-home / workspace and then these plugins could literally throw an error in the render when encountered to trigger this error banner view as a sort of IoC mechanism? Anyway this sounds like something that we might want to think through a bit more before trying to generalize 🤔